moabb/evaluations/evaluations.py

```diff
@@ -177,142 +181,108 @@ def _evaluate(
-                X_ = X[ix]
-                y_ = y[ix] if self.mne_labels else y_cv
+        name, clf = pipeline.items()
+        if _carbonfootprint:
+            # Initialize CodeCarbon
+            tracker = EmissionsTracker(save_to_file=False, log_level="error")
+            tracker.start()
+        t_start = time()
 
-                grid_clf = clone(clf)
+        # To-do: find a way to expose this for.
+        # Here, we will have n_splits = n_sessions*n_splits (default 5)
+        for cv_ind, (train_idx, test_idx) in enumerate(
+            self.cv.split(X, y, groups=metadata)
+        ):
+            session = metadata[train_idx][0]
+
+            # Create folder for grid search results
+            create_save_path(
+                self.hdf5_path,
+                dataset.code,
+                subject,
+                session,
+                name,
+                grid=True,
+                eval_type="WithinSession",
+            )
 
-                # Create folder for grid search results
-                create_save_path(
+            if self.hdf5_path is not None:
+                model_save_path = create_save_path(
                     self.hdf5_path,
                     dataset.code,
                     subject,
                     session,
                     name,
-                    grid=True,
+                    grid=False,
                     eval_type="WithinSession",
                 )
 
-                # Implement Grid Search
-                grid_clf = self._grid_search(
-                    param_grid=param_grid, name=name, grid_clf=grid_clf, inner_cv=inner_cv
-                )
-                if self.hdf5_path is not None:
-                    model_save_path = create_save_path(
-                        self.hdf5_path,
-                        dataset.code,
-                        subject,
-                        session,
-                        name,
-                        grid=False,
-                        eval_type="WithinSession",
-                    )
-
-                if isinstance(X, BaseEpochs):
-                    scorer = get_scorer(self.paradigm.scoring)
-                    acc = list()
-                    X_ = X[ix]
-                    y_ = y[ix] if self.mne_labels else y_cv
-                    for cv_ind, (train, test) in enumerate(cv.split(X_, y_)):
-                        cvclf = clone(grid_clf)
-                        cvclf.fit(X_[train], y_[train])
-                        acc.append(scorer(cvclf, X_[test], y_[test]))
-
-                        if self.hdf5_path is not None:
-                            save_model_cv(
-                                model=cvclf, save_path=model_save_path, cv_index=cv_ind
-                            )
+            scorer = get_scorer(self.paradigm.scoring)
+            acc = list()
+            cvclf = deepcopy(clf)
+            cvclf.fit(X[train_idx], y[train_idx])
+            acc.append(scorer(cvclf, X[test_idx], y[test_idx]))
```

Comment:
```suggestion
            le = LabelEncoder()
            y_train = le.fit_transform(y[train_idx])
            y_test = le.transform(y[test_idx])
            cvclf.fit(X[train_idx], y_train)
            acc.append(scorer(cvclf, X[test_idx], y_test))
```
Why not do simply this instead of encoding at the session level?